btcstaking-tracker/stakingeventwatcher/stakingeventwatcher.go

```diff
@@ -368,17 +370,44 @@ func tryParseStakerSignatureFromSpentTx(tx *wire.MsgTx, td *TrackedDelegation) (
 
 	stakingTxInput := tx.TxIn[stakingTxInputIdx]
 	witnessLen := len(stakingTxInput.Witness)
-	// minimal witness size for staking tx input is 4:
-	// covenant_signature, staker_signature, script, control_block
+	// minimal witness size for staking tx input is at least 4:
+	// covenant_signatures, staker_signatures, script, control_block
 	// If that is not the case, something weird is going on and we should investigate.
 	if witnessLen < 4 {
 		panic(fmt.Errorf("staking tx input witness has less than 4 elements for unbonding tx %s", tx.TxHash()))
 	}
 
-	// staker signature is 3rd element from the end
-	stakerSignature := stakingTxInput.Witness[witnessLen-3]
+	// staker signatures started from the 3rd element from the end
+	var (
+		schnorrSigs []*schnorr.Signature
+	)
+
+	// if it's a multisig btc delegation, parse signature on every non-nil stakerSignatures
+	if td.IsBtcMultisig {
+		stakerSignatures := stakingTxInput.Witness[witnessLen-3-int(td.StakerCount)+1 : witnessLen-3+1]
```

Comment:
can we add a comment on top of this making clear why the 3 and 1 hardcoded values? maybe put in a separated function